### DISCUSSION TITLE
using sequential forecast lstm to forecast future streamflow

### DISCUSSION THREAD

**Discussion by azadeh-jb** [neuralhydrology/neuralhydrology#156] [Q&A]
Hi,
I already used cuda lstm and its conf parameter (predict_last_n) and it worked perfectly! (thank you) . Now I'm trying to forecast the future streamflows using the **sequential forecast lstm**. I need to provide both predict_last_n AND forecast_seq_length in the config for this forecast lstm model (otherwise i get an error during training). 
My question is the results that I get after evaluation (eval_run) would be the future forecast ( t + "forecast_seq_length" ) or is it similar to cuda lstm where the result is (t- 'predict_last_n' ).

I guess I am a bit confused about need of both predict_last_n , forecast_seq_length parameters in config for sequential forecast lstm. Greatly appreciate your help.

Thanks,
Azadeh

**Comment by gauchm** [comment]
The two configuration parameters describe difference concepts:
- `forecast_seq_length` will determine the length of the sequence of forecast features that is fed into the model. 
- `predict_last_n` will determine how many time steps the loss is calculated on and how many time steps are returned during evaluation (counting from the end of the full sequence). I.e., if `predict_last_n > forecast_seq_length`, you will see a mix of "hindcasts" and "forecasts" in the results.

The logic is exactly the same as for non-forecast models, so the results are for time steps `t - predict_last_n` until `t`.

Finally, note that as described in the [docs](https://neuralhydrology.readthedocs.io/en/latest/usage/config.html):
> this does not currently support a forecast rollout, meaning that because forecast inputs behave the same way as dynamic inputs, the forecast input for two timesteps ahead of time t will be the same as the forecast input for one day ahead of time t+1.
